src/components/button/button.test.tsx

```diff
@@ -17,57 +17,91 @@ function sut(props: ButtonProps = defaultButton): HTMLElement {
+      );
+    });
 
-  afterEach(() => {
-    clickEvent.mockClear();
+    afterEach(() => {
+      clickEvent.mockClear();
+    });
   });
-});
 
-describe('Button Types', () => {
-  const buttonTypes: Array<ButtonProps['type']> = [
-    'primary',
-    'secondary',
-    'ghost',
-    'dashed',
-  ];
+  describe('Button Types', () => {
+    const buttonTypes: Array<ButtonProps['type']> = [
+      'primary',
+      'secondary',
+      'ghost',
+      'dashed',
+    ];
 
-  it.each(buttonTypes)('should render button with %s style type', (type) => {
-    const { className } = sut({ ...defaultButton, type: type });
-    expect(className).toContain(`type-${type}`);
+    it.each(buttonTypes)('should render button with %s style type', (type) => {
+      const { className } = sut({ ...defaultButton, type: type });
+      expect(className).toContain(`type-${type}`);
+    });
+
+    it.each(buttonTypes)('should render %s danger button', (type) => {
+      const { className } = sut({ ...defaultButton, type: type, danger: true });
+      expect(className).toContain('danger-true');
+      expect(className).toContain(`type-${type}`);
+    });
   });
 
-  it.each(buttonTypes)('should render %s danger button', (type) => {
-    const { className } = sut({ ...defaultButton, type: type, danger: true });
-    expect(className).toContain('danger-true');
-    expect(className).toContain(`type-${type}`);
+  describe('Button Sizes', () => {
+    const buttonSizes: Array<ButtonProps['size']> = ['sm', 'md', 'lg', 'xl'];
+
+    it.each(buttonSizes)(
+      'should render button with %s size variation',
+      (size) => {
+        expect(sut({ ...defaultButton, size: size }).className).toContain(
+          `size-${size}`
+        );
+      }
+    );
   });
-});
 
-describe('Button Sizes', () => {
-  const buttonSizes: Array<ButtonProps['size']> = ['sm', 'md', 'lg', 'xl'];
+  describe('With Icon', () => {
+    it('should render md icon size by default', () => {
+      const sm = '20';
+      const icon = 'pencil';
+      sut({ ...defaultButton, icon: icon });
+      const Tab = screen.getByTestId(`ion-icon-${icon}`);
+      expect(Tab).toHaveAttribute('height', sm);
+    });
 
-  it.each(buttonSizes)(
-    'should render button with %s size variation',
-    (size) => {
-      expect(sut({ ...defaultButton, size: size }).className).toContain(
-        `size-${size}`
+    it('should render icon lg when button is lg', () => {
+      const lgSize = '24';
+      const lgIconName = 'alert';
+      sut({
+        ...defaultButton,
+        name: 'button-icon',
+        size: 'lg',
+        icon: lgIconName,
+      });
+      expect(screen.getByTestId(`ion-icon-${lgIconName}`)).toHaveAttribute(
+        'height',
+        lgSize
       );
-    }
-  );
+    });
+
+    it('should render icon in right side', () => {
+      const icon = 'alert';
+      expect(
+        sut({ ...defaultButton, icon: icon, iconOnRight: true }).className
+      ).toContain('iconOnRight-true');
+    });
```

Comment:
I suggest create a spec to check if o icon is on left when the attr iconOnRight not is informed